Makefile

```diff
@@ -2,35 +2,44 @@ CPP          = g++
 CPPOPTS      = -Wall -Werror -std=c++11 -Iinclude/
 DEBUG_OPTS   = -fprofile-arcs -ftest-coverage -O0 -g -fPIC
 RELEASE_OPTS = -O3
-DEBUG_LIBS   = debug/url.o debug/utf8.o debug/punycode.o debug/psl.o
-RELEASE_LIBS = release/url.o release/utf8.o release/punycode.o release/psl.o
 
-release: $(RELEASE_LIBS)
+# Release libraries
+release:
+	mkdir -p release
+
+release/liburl.o: release/url.o release/utf8.o release/punycode.o release/psl.o
+	ld -r -o $@ $^
 
 release/%.o: src/%.cpp include/%.h
 	mkdir -p release
 	$(CPP) $(CPPOPTS) $(RELEASE_OPTS) -o $@ -c $<
 
-debug: $(DEBUG_LIBS)
+# Debug libraries
+debug:
+	mkdir -p debug
+
+debug/liburl.o: debug/url.o debug/utf8.o debug/punycode.o debug/psl.o
```

Comment:
You may want to simply switch to a static library representation using `ar` and `liburl.a` since that's a more standard representation for libraries.